components/SearchForm.js

```diff
@@ -49,15 +49,26 @@ class SearchForm extends React.Component {
 
   handleSubmit = event => {
     event.preventDefault();
-    const searchInput = event.target.elements.q;
-    this.setState({ isLoading: true });
-    this.props.router.push({ pathname: '/search', query: { q: searchInput.value } });
+    /*
+     * Here we set the loading state for a few seconds to show
+     * the loading spinner in place of the search button within the search bar.
+     */
+    this.setState({ isLoading: true }, () => {
+      setTimeout(() => {
+        this.setState({ isLoading: false });
+      }, 3000);
+    });
```

Comment:
What happens after 3 seconds ?

Reply:
After three seconds the button reverts back to normal state. That is the loading spinner appears for 3 secs and goes away. This is more of a visual effect than an actual loading state indication. Thinking further I feel we might as well remove this and only show the NProgress bar (there's nowhere as per design files that indicated we should use this spinner; I just did it based on this discussion; https://opencollective.slack.com/archives/C035S573ZD2/p1655212031150739?thread_ts=1655114037.455389&cid=C035S573ZD2). 🤔 Are you okay with that?

Reply:
I agree. I removed the spinner from the search form. I think until we find a permanent fix (maybe using NProgress itself) to show a spinner that syncs with NProgress bar we shouldn't add this spinner for the search form or the modal. I've added a commit to remove it.